pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/LabelScript.groovy

```diff
@@ -41,22 +41,34 @@ public class LabelScript extends DeclarativeAgentScript<Label> {
         return {
             try {
                 script.node(describable?.label) {
-                    if (describable.isDoCheckout() && describable.hasScmContext(script)) {
-                        if (!describable.inStage) {
-                            script.stage(SyntheticStageNames.checkout()) {
-                                script.checkout script.scm
-                            }
-                        } else {
-                            // No stage when we're in a nested stage already
-                            script.checkout script.scm
+                    if (describable.customWorkspace != null && describable.customWorkspace != "") {
```

Comment:
This is Groovy; you should be able to shorten it to `if (describable.customWorkspace)`